### DISCUSSION TITLE
Stato sonoff mini e relè passo passo

### DISCUSSION THREAD

**Discussion by essegi65** [indomus/forum#473] [GENERALE - Discussioni generiche]
Salve a tutti.
ho collegato un sono mini (con tasmota) ad un relè passo passo in modo che questo mi faccia da contatto pulito (ovvero quando attivo il sonoff mini questo mi deve eccitare la bobina del relè e chiudere il contatto).
Il mio problema è che non riesco una volta impostato sul sonoff mini il "pulsetime" a 1" (per non lasciare la bobina del relè sempre eccitata) ad avere lo stato del relè. in sostanza se attivo il sonoff mini questo eccita la bobina del relè il contatto dello stesso si chiude, ma il sonoff mini torna su OFF dopo 1". e quindi lo stato mi torna su off ovviamente mentre il relè è ON.
Potete aiutarmi o indicarmi dove istruirmi, grazie

**Comment by Larry-78** [comment]
Ciao
L'approccio secondo me è sbagliato, se posso darti un consiglio, dovresti utilizzare il contatto pulito del relè passo passo per far cambiare di stato il sonoff mini su S1 e S2 e di conseguenza sarà lui ad alimentare il tuo punto luce. In questo modo avrai pieno controllo e relativo stato ON/OFF

**Reply by Larry-78** [reply]
... dimenticavo una cosa importante, il relè deve essere di tipo Finder 26

**Reply by essegi65** [reply]
Scusami ma non avevo scritto che il contatto pulito del relè mi serve per dare il consenso ad un temporizzatore elettronico non domotico.

**Reply by Larry-78** [reply]
Allora le cose sono un po' diverse, ci sarebbero varie soluzioni ma visto che sei già in possesso di un mini, potresti trasformarlo a contatto pulito così da poter comandare direttamente il tuo temporizzatore.
C'è una guida, 
https://indomus.it/progetti/realizzare-un-contatto-pulito-modificando-un-sonoff-mini-r1/

**Reply by essegi65** [reply]
Il sonoff mini programmato con pulsetime a 2 secondi potrebbe essere una soluzione, ma lo stato come lo controllo?

**Reply by Larry-78** [reply]
Il temporizzatore per partire ha bisogno di un impulso oppure necessita di un contatto pulito da cortocircuitare per il tempo necessario? Se bastasse solo un impulso, non puoi utilizzare un relè passo passo e non potrai mai ottenere il controllo dello stato con pulsetime. Dovresti cambiare tipologia e utilizzare un relè monostabile così quando il pulsetime porterà a OFF il mini, il relè non sarà più eccitato e la bobina aprirà il contatto.

**Reply by essegi65** [reply]
Al temporizzatore occorre un contatto pulito e serve un pulsante per comandare la bobina del relè. Il mio relè una volta dato il consenso alla bobina chiude (o apre il contatto) e la bobina NON rimane eccitata. il relè è [questo](https://www.amazon.it/gp/product/B00PM1ECLU/ref=ppx_yo_dt_b_asin_title_o05_s01?ie=UTF8&th=1)

**Reply by Larry-78** [reply]
Ok confermo che non è adatto per essere comandato da un sonoff mini. Se non vuoi modificarlo a contatto pulito, ci sarebbero svariate soluzioni, sempre se il tuo scopo è ottenere lo stato su HA. Tra le quali potresti utilizzare per esempio uno Shelly 1 che ha un contatto pulito oppure una scheda nodemcu con tasmota che comanda una scheda relè a 5V oppure cercare di realizzare un temporizzatore direttamente su HA (quello in tuo possesso a quale dispositivo è collegato?)

**Reply by essegi65** [reply]
ok visto che ho uno shelly 1 per un altro utilizzo faccio un cambio con il sonoff mini, grazie di tutto

**Reply by daiv87** [reply]
Domanda … perché sbatterti per tutto questo lavoro quando potresti bypassare il relè con il sonoff? Mettendo tasmota/esphome puoi sfruttare delle rules interne e fare da temporizzatore il sonoff stesso.
E poi se non ricordo male mettendo un relè a di tipo passo passo non vorrei essere in errore ma sicuro che puoi avere lo stato?

**Reply by essegi65** [reply]
Grazie per i tuoi consigli. Ma per il temporizzatore (che fa parte di una macchina) mi occorre avere la segnalazione su una sicurezza aperta o chiusa (quindi con contatto pulito)e il sonoff mini (già con tasmota) che avevo non lo volevo modificare a livello di hardware. Ho risolto con lo shelly.

**Reply by daiv87** [reply]
Ottimo ti sei levato la rogna della modifica hw sul mini che essendo piccolo sarebbe stato una bella sfida.
